code/datums/components/_component.dm

```diff
@@ -14,8 +14,7 @@
 	var/list/arguments = args.Copy()
 	arguments.Cut(1, 2)
 	if(Initialize(arglist(arguments)) == COMPONENT_INCOMPATIBLE)
-		parent = null
-		qdel(src)

```

Comment:
problem with this is parent is set to null to prevent _RemoveFromParent() from being called in this case (because it's not IN the parent yet) this needs to be maintained somehow